### PR TITLE
Added missing dependencies to themes/pom.xml

### LINKED ISSUE #24571: Parallel builds stopped working
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

Using parallel build to compile Keycloak has stopped working between 21.0.0 and 22.0.0 releases.

### Version

main

### Expected behavior

Compilation should succeed.

### Actual behavior

Compiling fails with errors

```
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-js-parent: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-js-admin-client: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-account-ui: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-admin-ui: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
```

### How to Reproduce?

Compile Keycloak by running

```
mvn -T4C clean install -DskipTestsuite -DskipExamples -DskipTests
```

### Anything else?

I've used `git bisect` to track the commit which makes parallel builds fail and it seems to be #17401 and commit  https://github.com/keycloak/keycloak/commit/6d2e57f93a297360c4ebb6082dcd7fc87ccf2448.

### PR DESCRIPTION
**Pull request by tsaarni**
This PR adds dependency to `keycloak-admin-ui` and `keycloak-account-ui` modules. Both modules contribute files to `themes` and are therefore dependencies for `themes`

Closes #24571, but only with the precondition that #24537 is also merged and it would resolve concurrent node and pnpm install execution https://github.com/keycloak/keycloak/pull/24537#discussion_r1389486794

### CODE REVIEW

**Comment by tsaarni** [comment]
Cc @ASzc @jonkoops

File: themes/pom.xml [hunk 1]
```diff
@@ -22,6 +22,19 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.keycloak</groupId>
+            <artifactId>keycloak-admin-ui</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.keycloak</groupId>
+            <artifactId>keycloak-account-ui</artifactId>
+            <version>${project.version}</version>
```
Comment (ASzc) [inline themes/pom.xml:34]: A `<version>` is not required, because these two modules have versions specified in `keycloak-parent`'s dependencyManagement

**Review by ASzc — CHANGES_REQUESTED** [review]
Seems to make sense to me. I imagine the reason there isn't a dependency defined already is because of optimization. The poms have have several instances of broken maven metadata, especially where the javascript stuff is concerned, which I'm not overly happy with, but would slightly speed up a build

**Review by jonkoops — APPROVED** [review]
I have no objection against merging this in, but we do need a maintainer to take a look at this.

**Review by ahus1 — APPROVED** [review]
Thank you, the changes make sense.

**Pull request merged**